hadoop-hdds/annotations/pom.xml

```diff
@@ -34,4 +34,16 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
   <properties>
     <maven.test.skip>true</maven.test.skip> <!-- no tests in this module so far -->
   </properties>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
```

Comment:
why do we need to add the maven-compiler-plugin explicitly in every pom file even when we do not configure the path for an annotation processor?

Reply:
I would guess this is for number 2. `limit the list of processors, or even disable processing completely, based on each module's needs.
`

Reply:
Yes, this case is for "disable processing".

I guess we could make `<proc>none</proc>` the default by adding it in root POM, and overriding where necessary.

Reply:
Unfortunately Java 8 doesn't provide any way to override `<proc>none</proc>`, because the default ("process and compile") has no corresponding option.  Later Java versions can use `<proc>full</proc>`.

So we can simplify this only after dropping Java 8 support.